pandas/tests/tseries/holiday/test_federal.py

```diff
@@ -36,3 +39,15 @@ class MemorialDay(AbstractHolidayCalendar):
         datetime(1978, 5, 29, 0, 0),
         datetime(1979, 5, 28, 0, 0),
     ]
+
+
+def test_federal_holiday_inconsistent_returntype():
+    # GH 49075 test case
+    # Instantiate two calendars to rule out _cache
+    cal1 = USFederalHolidayCalendar()
+    cal2 = USFederalHolidayCalendar()
+
+    results_2018 = cal1.holidays(start=datetime(2018, 8, 1), end=datetime(2018, 8, 31))
+    results_2019 = cal2.holidays(start=datetime(2019, 8, 1), end=datetime(2019, 8, 31))
+
+    tm.assert_index_equal(results_2018, results_2019)
```

Comment:
One last item: Could you assert that `results_2018` and `results_2019` equal manually constructed `DatetimeIndex`? Just in case a future change affects both in bad way

Reply:
- Replaced the comparisons here with separate comparisons against a constructed `DatetimeIndex`
- Cleaned up the longer checks in `test_holiday` and replaced with comparison against constructed `DatetimeIndex`

Ran precommit, fixed `black`/`isort`/`flake8` warnings, re-ran `pre-commit`, and off we go.